clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -75,6 +75,28 @@ def filter_dicoms(df: DataFrame) -> DataFrame:
     return df
 
 
+def _handle_series_description(x: str) -> str:
+    """This function handles the series description used to identify the modality later on.
+
+    In the  case of T1 and T2, some dicoms are misnamed. Therefore we use the echo time and repetition time
+    which are supposed to be very different for a T1 and T2, to check the series description corresponds to what has been done.
+    The values used are in milliseconds, and have been chosen arbitrarily.
+    """
+    MAX_ECHO_TIME_FOR_A_T1 = 60
+    MAX_REPETITION_TIME_FOR_A_T1 = 2100
+    series_description = pdcm.dcmread(x).SeriesDescription
+    if any([modality in series_description.lower() for modality in ["t1", "t2"]]):
+        echo_time = pdcm.dcmread(x).EchoTime
+        repetition_time = pdcm.dcmread(x).RepetitionTime
+        if (
+            "t2" in series_description.lower()
+            and echo_time < MAX_ECHO_TIME_FOR_A_T1
+            and repetition_time < MAX_REPETITION_TIME_FOR_A_T1
+        ):
+            return "t1"
```

Comment:
Is it an issue that the function returns only "t1" in this case, but the full series description in other cases ?

Reply:
No it is not. The function `identify_modality` looks for t1 in the series description